### PR TITLE
precice: update

### PR DESCRIPTION
**Pull request by 3rav**
Update PKGBUILD

Update PKGBUILD

Update PKGBUILD

### CODE REVIEW

**Comment by Biswa96** [comment]
The binprecice executable is installed as symbolic link ([ci log][1]). Shouldn't it be installed as a copy of precice-tool.exe? symbolic link creation requires admin privilege if developer mode is not enabled, right?

[1]: https://github.com/msys2/MINGW-packages/runs/6494750829?check_suite_focus=true#step:9:1457

**Comment by lazka** [comment]
yeah, that needs to be fixed somehow. installing the package will fail without symlink support.

**Comment by fsimonis** [comment]
Hi, this issue is tricky to fix as there is no copy_on_error option of the CMake CLI version of the `create_symlink` command.

We will release one more minor version before version 3. So, this symlink will not stick around for much longer.

I recommend to simply replace it by a `copy` or `copy_if_different`.

```diff
   add_custom_target(
     binprecice
     ALL
-    COMMAND ${CMAKE_COMMAND} -E create_symlink precice-tools binprecice
+    COMMAND ${CMAKE_COMMAND} -E copy_if_different precice-tools binprecice
     DEPENDS precice-tools
     WORKING_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}
     BYPRODUCTS binprecice
     )
```

File: mingw-w64-precice/PKGBUILD [hunk 1]
```diff
@@ -26,7 +26,12 @@ makedepends=("${MINGW_PACKAGE_PREFIX}-petsc-build"
 license=('LGPL3')
 url='https://www.precice.org'
 source=(https://github.com/precice/precice/archive/v${pkgver}.tar.gz)
-sha256sums=('57BAB08E8B986F5FAA364689D470940DBD9C138E5CFA7B861793E7DB56B89DA3')
+sha256sums=('762E603FBCAA96C4FB0B378B7CB6789D09DA0CF6193325603E5EEB13E4C7601C')
+
+prepare() {
+  cd "${_realname}-${pkgver}"
+  sed -i 's/BYPRODUCTS binprecice/#BYPRODUCTS binprecice/' CMakeLists.txt
```
Comment (fsimonis) [inline mingw-w64-precice/PKGBUILD:33]: This should work for packages until version 3

```suggestion
  sed -i 's/E create_symlink/E copy_if_different/' CMakeLists.txt
```

File: mingw-w64-precice/PKGBUILD [hunk 2]
```diff
@@ -26,7 +26,14 @@ makedepends=("${MINGW_PACKAGE_PREFIX}-petsc-build"
...
+  #Temporary fix: https://github.com/msys2/MINGW-packages/pull/11685#issuecomment-1200834235
+  sed -i 's/E create_symlink/E copy_if_different/' CMakeLists.txt
```
Comment (mmuetzel) [inline mingw-w64-precice/PKGBUILD:35]: You might need to include the exe extension here:
``` diff
-  sed -i 's/E create_symlink/E copy_if_different/' CMakeLists.txt
+  sed -i 's/E create_symlink precice-tools binprecice/E copy_if_different precice-tools${CMAKE_EXECUTABLE_SUFFIX} binprecice${CMAKE_EXECUTABLE_SUFFIX}/' CMakeLists.txt
```

And also rebase on a current head of the master branch.

**Pull request closed without merging**